src/lib/corchuelo.rs

```diff
@@ -253,7 +253,7 @@ impl<TokId: Clone + Copy + Debug + TryFrom<usize> + TryInto<usize> + PartialEq>
             for r in repairs[0].iter() {
                 match *r {
                     Repair::InsertTerm{term_idx} => {
-                        let (next_lexeme, _) = parser.next_lexeme(None, la_idx);
```

Comment:
I didn't get this immediately until I looked at the second commit and reaslied that `next_lexeme` is only introduced there, which reverts these changes again. Maybe merging the two commits would actually make things easier to understand for once.

Reply:
Up to you. I'm probably marginally inclined to keep them separate, but I don't feel strongly about it... and I'm a little biased!

Reply:
Alternatively you could also just leave this line `let (next_lexeme, _) = parser.next_lexeme(None, la_idx);` as is. The change to reactor this to `.0` only to remove parts of it again in the next commit seems a bit pointless. Unless I am missing something. But I'm not bothered enough to make you rewrite your commits. So if you think it should stay then I'm happy to merge.

Reply:
Just this one line? If so, I'll rebase and squash at the same time.

Reply:
Nah I think it's a couple of lines. You are changing
```
- let (next_lexeme, _) = parser.next_lexeme(None, la_idx);
+ let next_lexeme = parser.next_lexeme(None, la_idx).0;
```
in the first commit. Then in the second you are changing
```
- let next_lexeme = parser.next_lexeme(None, la_idx).0;
+ let next_lexeme = parser.next_lexeme(la_idx);
```
It seems you could have just changed this in the second commit only:
```
- let (next_lexeme, _) = parser.next_lexeme(None, la_idx);
+ let next_lexeme = parser.next_lexeme(la_idx);
```
Maybe I'm wrong. Or maybe it's a bit nitpicky. I was just a bit confused when I first read the commits, so I thought I point it out.

Reply:
You're right. Let me see how tricky this is to fix.